ui/src/composables/use-notifications.js

```diff
@@ -0,0 +1,114 @@
+import asteroidConfig from 'asteroid-config'
+import hasParentByClassName from '../helpers/private/has-parent-by-class-name'
+
+import { Notify } from 'quasar'
+import { ref } from 'vue'
+
+const callbackFunctions = {
+  onNotificationReceived: []
+}
+
+export function onNotificationReceived (callbackFn) {
+  callbackFunctions.onNotificationReceived.push(callbackFn)
+}
+
+const unreadNotificationsCount = ref(0)
+
+export default function () {
+  const isNotificationsEnabled = asteroidConfig.framework.featureToggle?.useNotifications
+
+  function triggerNotification (notification) {
+    callbackFunctions.onNotificationReceived.forEach(fn => fn((notification)))
+  }
+
+  function setUnreadNotificationsCount (value) {
+    unreadNotificationsCount.value = value
+  }
+
+  function incrementUnreadNotificationsCount () {
+    unreadNotificationsCount.value++
+  }
+
+  /**
+   * @param {{
+   *  message: string
+   *  title: string
+   *  link?: string
+   * }} config
+   */
+  function sendNotify (config = {}) {
+    const { link } = config
+
+    const classes = ['bg-white', 'boot-notification', 'q-py-sm', 'text-grey-8']
+
+    if (link) classes.push(...['boot-notification--has-link', 'cursor-pointer'])
+
+    const closeNotify = Notify.create({
+      actions: [{
+        icon: 'sym_r_close',
+        class: 'boot-notification__close-button qas-btn qas-btn--tertiary qas-btn--tertiary-grey-10'
+      }],
+
+      attrs: {
+        onClick: event => onNotifyClick({ event, link, closeNotify })
+      },
+
+      classes: classes.join(' '),
+      html: true,
+      icon: 'sym_r_info',
+      iconColor: 'primary',
+      message: getHTMLMessage(config),
+      multiLine: true,
+      position: 'bottom-right',
+      timeout: 30000
+    })
+
+    /**
+     * Função que é chamada quando o usuário clica na notificação, se a notificação
+     * tem link, então ele vai ser redirecionado para o link em uma nova aba, caso
+     * não tenha link, nada acontece.
+     */
+    function onNotifyClick ({ event, link, closeNotify }) {
+      if (!link) return
+
+      if (hasParentByClassName('boot-notification__close-button', event.srcElement)) return
+
+      window.open(link)
```

Comment:
Se for um link externo não precisa tratar pra abrir em uma nova aba?

Reply:
Neste caso SEMPRE que tiver link vai ser redirecionado pra outra pagina... se não tem link nunca é redirecionado, nada acontece